rpc-web3js: Allow passing Connection class directly

mPaella opened the pull request:
we would like to be able to pass our own class, which itself implements `Connection`, rather than only being able to pass a single rpc endpoint or having to reimplement our custom `Connection` logic for the `RpcInterface` interface

- allow passing `Connection` to `createUmi` from `umi-bundle-defaults`
- allow passing `getAccountsChunkSize` to `createUmi` from `umi-bundle-defaults`
- allow passing `Connection` to `createWeb3JsRpc` and `web3JsRpc` from `umi-rpc-web3js`

mPaella commented on packages/umi-bundle-defaults/src/index.ts:
```diff
@@ -1,10 +1,26 @@
+): Umi {
+  return baseCreateUmi().use(
+    typeof endpointOrConnection === 'string'
+      ? defaultPlugins(endpointOrConnection, rpcOptions)
+      : defaultPlugins(endpointOrConnection, rpcOptions)
```
this is slightly weird since its the same function call params, but ts doesnt like it without the ternary

![image](https://github.com/metaplex-foundation/umi/assets/93682696/740f49c5-bad1-4877-9ca2-7fd6b4876c66)

lorisleiva commented:
I like that, thanks! Am I right to say this change does not introduce any breaking change to the API? If so, could you please add a `patch` changeset?

mPaella commented:
> I like that, thanks! Am I right to say this change does not introduce any breaking change to the API? If so, could you please add a `patch` changeset?

correct!

added changeset, and updated the docs

lorisleiva commented on docs/rpc.md:
````diff
@@ -4,13 +4,19 @@ Contacting the Solana blockchain via an RPC is an important part of any decentra
-When creating a new Umi instance via the default bundle, you must pass the RPC's endpoint as the first argument. Going forward, this it the endpoint that will be used every time you call a method on the RPC interface.
+When creating a new Umi instance via the default bundle, you must pass the RPC's endpoint or an instance of `@solana/web3.js`'s `Connection` class as the first argument. Going forward, this is the endpoint or `Connection` that will be used every time you call a method on the RPC interface.
 
 ```ts
 import { createUmi } from '@metaplex-foundation/umi-bundle-defaults';
 
 const umi = createUmi("https://api.mainnet-beta.solana.com");
 ```
+```ts
+import { createUmi } from '@metaplex-foundation/umi-bundle-defaults';
+import { Connection } from '@solana/web3.js';
+
+const umi = createUmi(new Connection("https://api.mainnet-beta.solana.com"));
+```
````
Could you just merge these two code blocks (and the ones below) together. Having two code blocks without text in between them looks a bit ugly. I suggest the follow instead:

```ts
import { createUmi } from '@metaplex-foundation/umi-bundle-defaults';
import { Connection } from '@solana/web3.js';

// Pass in your RPC endpoint.
const umi = createUmi("https://api.mainnet-beta.solana.com");

// Or an explicit Connection object from web3.js.
const umi = createUmi(new Connection("https://api.mainnet-beta.solana.com"));
```

Also note that, eventually the `Connection` object from the legacy web3.js library will be a thing of the past so this API will need to be updated but that's fine for now.

mPaella replied:
updated

makes sense, excited for that day :) do you think umi will end up using/extending the new rpc interface from web3.js@experimental, or it will continue to have its own interface?

lorisleiva replied:
Thanks! I'll definitely end up relying a lot more on the new web3.js for types and helpers so that both of them are compatible and can be used together effortlessly. But also I want to keep the ease of use that some of the current interfaces offer.

lorisleiva reviewed (approved):
Thanks! 🍻
